### PR TITLE
Exposing the transport window buffer sizes 

### PR DESCRIPTION
**Pull request by improbable-danny**
#### Description
Configurable transport window sizes

#### Release note
Exposing worker upstream/downstream window sizes as GDK options for both clients and servers, (`ClientDownstreamWindowSizeBytes`, `ClientUpstreamWindowSizeBytes`) and (`ServerDownstreamWindowSizeBytes` and `ServerUpstreamWindowSizeBytes`).

#### Primary reviewers

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 1]
```diff
@@ -147,6 +155,8 @@ struct FConnectionConfig
 	uint8 TcpNoDelay;
 	uint8 UdpUpstreamIntervalMS;
 	uint8 UdpDownstreamIntervalMS;
+	uint16 DownstreamWindowSize;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h:158]: uint32 ?

**Comment by improbable-danny** [comment]
* Add a release note

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 2]
```diff
@@ -98,7 +108,8 @@ struct FConnectionConfig
 		}
 		else if (!LogLevelString.IsEmpty())
 		{
-			UE_LOG(LogConnectionConfig, Warning, TEXT("Unknown worker SDK log verbosity %s specified. Defaulting to Info."), *LogLevelString);
+			UE_LOG(LogConnectionConfig, Warning, TEXT("Unknown worker SDK log verbosity %s specified. Defaulting to Info."),
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h:111]: This was auto formatting

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 3]
```diff
@@ -73,6 +73,16 @@ struct FConnectionConfig
 													: SpatialGDKSettings->UdpServerDownstreamUpdateIntervalMS);
 
 		LinkProtocol = ConnectionTypeMap[bConnectAsClient ? EWorkerType::Client : EWorkerType::Server];
+		static_assert(EWorkerType::Client == 0 && EWorkerType::Server == 1, "Assuming indexes of enum for client and server");
+
+		uint32 DownstreamWindowSizes[2] = { SpatialGDKSettings->ClientDownstreamWindowSizeBytes,
+											SpatialGDKSettings->ServerDownstreamWindowSizeBytes };
+		uint32 UpstreamWindowSizes[2] = { SpatialGDKSettings->ClientUpstreamWindowSizeBytes,
+										  SpatialGDKSettings->ServerUpstreamWindowSizeBytes };
+		static_assert(EWorkerType::Client == 0 && EWorkerType::Server == 1, "Assuming indexes of enum for client and server");
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h:82]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h [hunk 4]
```diff
@@ -147,6 +158,8 @@ struct FConnectionConfig
 	uint8 TcpNoDelay;
 	uint8 UdpUpstreamIntervalMS;
 	uint8 UdpDownstreamIntervalMS;
+	uint32 DownstreamWindowSize;
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/ConnectionConfig.h:161]: Add bytes to these

**Review by m-samiec — APPROVED** [review]
Couple of nits

**Pull request merged**